android/src/main/kotlin/com/airship/flutter/AirshipPlugin.kt

```diff
@@ -131,6 +133,69 @@ class AirshipPlugin : MethodCallHandler {
         result.success(null)
     }
 
+    private fun addEvent(call: MethodCall, result: Result) {
+        val eventMap = call.arguments as HashMap<*, *>
+
+        val eventName = eventMap[CustomEvent.EVENT_NAME] as String? ?: run {
+            result.success(false)
+            return
+        }
+
+        val event = CustomEvent.Builder(eventName).apply {
+            (eventMap[CustomEvent.EVENT_VALUE] as Int?)?.let {
+                this.setEventValue(it)
+            }
+
+            (eventMap[CustomEvent.PROPERTIES] as HashMap<String, Any>?)?.let {
+                parseProperties(it, this)
```

Comment:
I don't know Kotlin, but it seems like you could just use `this` inside `parseProperties()`, rather than passing it in?

Reply:
not able to do that in the described situation as the function's 'this' would not be the same as 'this' inside the let block, updated this to an extension though which cleans it up